test_vm/tests/verified_claim_test.rs

```diff
@@ -298,3 +300,94 @@ fn verified_claim_scenario() {
         &[invariant_failure_patterns::REWARD_STATE_EPOCH_MISMATCH.to_owned()],
     );
 }
+
+#[test]
+fn expired_allocations() {
+    let store = MemoryBlockstore::new();
+    let mut v = VM::new_with_singletons(&store);
+    let addrs = create_accounts(&v, 3, TokenAmount::from_whole(10_000));
+    let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P1;
+    let (owner, worker, verifier, verified_client) = (addrs[0], addrs[0], addrs[1], addrs[2]);
+
+    // Create miner
+    let (miner_id, _) = create_miner(
+        &mut v,
+        owner,
+        worker,
+        seal_proof.registered_window_post_proof().unwrap(),
+        TokenAmount::from_whole(1_000),
+    );
+    let v = v.with_epoch(200);
+
+    // Register verifier and verified clients
+    let datacap = StoragePower::from(32_u128 << 40);
+    verifreg_add_verifier(&v, verifier, &datacap * 2);
+    verifreg_add_client(&v, verifier, verified_client, datacap.clone());
+
+    // Add market collateral for client and miner
+    market_add_balance(&v, verified_client, verified_client, TokenAmount::from_whole(3));
+    market_add_balance(&v, worker, miner_id, TokenAmount::from_whole(64));
+
+    // Publish 2 verified deals
+    let deal1_start =
+        v.get_epoch() + max_prove_commit_duration(&Policy::default(), seal_proof).unwrap();
+    let deal_term_min = 180 * EPOCHS_IN_DAY;
+
+    let deal_size = 32u64 << 30;
+    let deal1 = market_publish_deal(
+        &v,
+        worker,
+        verified_client,
+        miner_id,
+        "deal1".to_string(),
+        PaddedPieceSize(deal_size),
+        true,
+        deal1_start,
+        deal_term_min,
+    )
+    .ids[0];
+
+    // Client datacap balance reduced
+    let datacap_state: DatacapState = v.get_state(DATACAP_TOKEN_ACTOR_ADDR).unwrap();
+    assert_eq!(
+        TokenAmount::from_whole(datacap.clone()) - TokenAmount::from_whole(deal_size),
+        datacap_state.token.get_balance(&v.store, verified_client.id().unwrap()).unwrap()
+    );
+
+    // Advance to after the first deal's start
+    let v = v.with_epoch(deal1_start + EPOCHS_IN_DAY);
```

Comment:
nit: `DEAL_UPDATES_INTERVAL` is slightly clearer and safer against policy changes